syntax = "proto3";

// Transmits values from the Teensy to the ESP32
message LSlaveToMaster {
    // Processed light sensor values
    float lineAngle = 1;
    float lineSize = 2;
    bool onLine = 3;
    bool lineOver = 4;
    float lastAngle = 5;

    // Raw LRF values
    int32 frontLRF = 6;
    int32 rightLRF = 7;
    int32 backLRF = 8;
    int32 leftLRF = 9;
}

// Transmits values from the ESP32 to the Teensy
message MasterToLSlave {
    // IMU
    float heading = 1;

    // Debug LEDs
    repeated bool debugLEDs = 2;
}

// Transmits values from the ATMega to the ESP32
message MSlaveToMaster {
    // Mouse sensor
    int32 mouseDX = 1;
    int32 mouseDY = 2;
}

// Transmits values from ESP32 to ATMega
message MasterToMSlave {
    // Motor outputs
    int32 frMotor = 1;
    int32 brMotor = 2;
    int32 blMotor = 3;
    int32 flMotor = 4;
}